translator/translate/otel/processor/jmxprocessor/translator.go

```diff
@@ -0,0 +1,106 @@
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+)
+
+const (
+	// Match types are in internal package from contrib
+	// Strict is the FilterType for filtering by exact string matches.
+	regexp = "regexp"
+)
+
+var (
+	jmxKey           = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey)
+	activeMqKey      = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.ActiveMqKey)
+	cassandraKey     = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.CassandraKey)
+	hbaseKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.HbaseKey)
+	hadoopKey        = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.HadoopKey)
+	jettyKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.JettyKey)
+	jvmKey           = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.JvmKey)
+	kafkaKey         = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaKey)
+	kafkaConsumerKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaConsumerKey)
+	kafkaProducerKey = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.KafkaProducerKey)
+	solrKey          = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.SolrKey)
+	tomcatKey        = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.TomcatKey)
+	wildflyKey       = common.ConfigKey(common.MetricsKey, common.MetricsCollectedKey, common.JmxKey, common.WildflyKey)
+
+	jmxTargets = []string{"activemq", "cassandra", "hbase", "hadoop", "jetty", "jvm", "kafka", "kafka-consumer", "kafka-producer", "solr", "tomcat", "wildfly"}
+)
+
+type translator struct {
+	name    string
+	factory processor.Factory
+}
+
+var _ common.Translator[component.Config] = (*translator)(nil)
+
+func NewTranslator() common.Translator[component.Config] {
+	return NewTranslatorWithName("")
+}
+
+func NewTranslatorWithName(name string) common.Translator[component.Config] {
+	return &translator{name, filterprocessor.NewFactory()}
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+// Translate creates a processor config based on the fields in the
+// Metrics section of the JSON config.
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	if conf == nil || !conf.IsSet(jmxKey) {
+		return nil, &common.MissingKeyError{ID: t.ID(), JsonKey: fmt.Sprint(jmxKey)}
+	}
+
+	cfg := t.factory.CreateDefaultConfig().(*filterprocessor.Config)
+
+	// If no target system or no filter is given, then the metric name should be
```

Comment:
nit: This comment seems incomplete. What is this trying to check?

Reply:
If no target system or no filter is given, the metric is supposed to be allowed by defaulted but @sethAmazon can confirm too

Reply:
Yes if there is no filter we should allow all from JXM

Reply:
The JMX receiver (https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/jmxreceiver/config.go#L227) won't work without a target system.